arrow-fx-test/src/main/kotlin/arrow/test/laws/Law.kt

```diff
@@ -24,7 +24,7 @@ fun <A> A.shouldBeEq(b: A, eq: Eq<A>): Unit = this should matchUnderEq(eq, b)
 
 fun <A> matchUnderEq(eq: Eq<A>, b: A) = object : Matcher<A> {
   override fun test(value: A): Result {
-    return io.kotlintest.Result(eq.run { value.eqv(b) }, "Expected: $b but found: $value", "$b and $value should be equal")
+    return Result(value.equalUnderTheLaw(b, eq), "Expected: $b but found: $value", "$b and $value should be equal")
```

Comment:
@nomisRev I actually did the opposite, make `shouldBeEq` above depend on `equalUnderTheLaw`

Reply:
The problem is that the `forAll` tests expect a boolean at the end, which is provided by the `equalUnderTheLaw`, otherwise for a explicit check or throw the `shouldXXX` should be used due to convention.

Reply:
Not really, since that was already the case. `equalUnderTheLaw == eq.run { eqv(b) }` and you substituted it in `matchUnderEq` which means nothing changed. Subsitution law ;)

Reply:
```kotlin
fun <A> A.equalUnderTheLaw(b: A, eq: Eq<A>): Boolean =
  this should matchUnderEq(eq, b).let { true }

fun <A> A.shouldBeEq(b: A, eq: Eq<A>): Unit = this should matchUnderEq(eq, b)

fun <A> matchUnderEq(eq: Eq<A>, b: A) = object : Matcher<A> {
  override fun test(value: A): Result {
    return io.kotlintest.Result(eq.run { value.eqv(b) }, "Expected: $b but found: $value", "$b and $value should be equal")
```

Reply:
This will make sure that all failing `forAll` fail with a proper message `"Expected: $b but found: $value", "$b and $value should be equal"`, instead of currently `"Expected: true but found: false`.